### ISSUE TITLE
`Jetpack_Options` not found in cloned installation

### ISSUE THREAD

**Issue by flowl** [Automattic/jetpack#20296]
I am cloning a friends WP installation using `mysqldump`, `rsync` and `docker` to fetch all files and data.

When running the site, I get the following error:

    Fatal error:  Uncaught Error: Class 'Jetpack_Options' not found in /var/www/html/wp-content/plugins/jetpack/class.jetpack.php:2154
    Stack trace:
    #0 /var/www/html/wp-content/plugins/jetpack/class.jetpack.php(702): Jetpack::get_option('edit_links_caly...')
    #1 /var/www/html/wp-content/plugins/jetpack/class.jetpack.php(403): Jetpack-&gt;__construct()
    #2 /var/www/html/wp-content/plugins/jetpack/load-jetpack.php(89): Jetpack::init()
    #3 /var/www/html/wp-content/plugins/jetpack/jetpack.php(149): require_once('/var/www/html/w...')
    #4 /var/www/html/wp-settings.php(371): include_once('/var/www/html/w...')
    #5 /var/www/html/wp-config.php(90): require_once('/var/www/html/w...')
    #6 /var/www/html/wp-load.php(37): require_once('/var/www/html/w...')
    #7 /var/www/html/wp-blog-header.php(13): require_once('/var/www/html/w...')
    #8 /var/www/html/index.php(17): require('/var/www/html/w...')
    #9 {main}
    thrown in <b>/var/www/html/wp-content/plugins/jetpack/class.jetpack.php</b> on line 154

The class exists, it is not missing, so I think this is an autoloading problem.

I tried to fix the problem by navigating into `/var/www/html/wp-content/plugins/jetpack/`, installing composer, and running `composer dump-autoload`.

    /var/www/html/wp-content/plugins/jetpack # php /composer.phar dump-autoload
    Generating autoload files
    Generated autoload files

This doesn't fix the problem.

How do I fix the classmaps of a mirrored wordpress codebase?

**Comment by jeherve** [comment]
Thanks for the report! That's an interesting problem.

Could you let us know how the original codebase was set up? Was the Jetpack plugin pulled via Packagist and built locally? If so, from which source and version?

Thank you!

**Comment by flowl** [comment]
@jeherve The installation on my friends server is setup through the Wordpress Admin page, nothing has been manually wired.
The installation is properly working on his server.

When I pull the files into my container, the autoloading of Jetpack only works partially:

`class.jetpack.php` file exists and can be loaded
`class-jetpack-options.php` file exists but doesn't get loaded

    wp-content/plugins/jetpack/vendor/automattic/jetpack-options/legacy/class-jetpack-options.php

I think the paths on the server and the paths on my container differ.
The problem should be resolved by updating composers autoload files, but it doesn't.


There are autoloading files in `wp-content/plugins/jetpack/vendor/composer` **and** `wp-content/plugins/jetpack/vendor/jetpack-autoloader`.

Which of those is correct and how do I properly regenerate the classmaps for the legacy class?

**Comment by jeherve** [comment]
> There are autoloading files in wp-content/plugins/jetpack/vendor/composer and wp-content/plugins/jetpack/vendor/jetpack-autoloader.

Both directories are useful, but you should only find classmaps in `vendor/composer`. You can see the latest production version of the plugin here:
- https://plugins.svn.wordpress.org/jetpack/tags/9.9/vendor/composer/
- https://github.com/Automattic/jetpack-production/tree/9.9/vendor

I'm not sure how the classmaps may not be valid on your end, or not match the location of the files if you've copied the whole thing as is.

What do you see for the Jetpack options file in the classmaps, before and after you dump autoload? Do you see anything different from the production version of the plugin?
- https://github.com/Automattic/jetpack-production/blob/9.9/vendor/composer/autoload_classmap.php#L112
- https://github.com/Automattic/jetpack-production/blob/9.9/vendor/composer/jetpack_autoload_classmap.php#L487

**Comment by jeherve** [comment]
Closing this for now because of the lack of activity on this. We can always reopen in the future if needed.

**Comment by flowl** [comment]
Hey @jeherve 
sorry I didn't reply yet!

I compared your files to mine and noticed a big difference, I copied the contents of your files into mine and it still didn't work as expected. I ended up removing the jetpack folder because we don't use the plugin anyways.

As I cannot provide a reproducible case, I suspect something went wrong on my side, outside of jetpack scope.

I very appreciate you taking time and your reply was helpful.
The ticket can be closed.

Wishing all the best
